tests/Functional/ContentPublicationRenderTest.php

```diff
@@ -0,0 +1,281 @@
+    $node->set('oe_departments', 'http://publications.europa.eu/resource/authority/corporate-body/ABEC');
+    $node->save();
+    $this->drupalGet($node->toUrl());
+
+    $details_expected_values['items'][] = [
+      'label' => 'Related department',
+      'body' => 'Audit Board of the European Communities',
+    ];
+    $field_list_assert->assertPattern($details_expected_values, $content_items[0]->getHtml());
+
+    // Assert multiple Related department field.
+    $node->set('oe_departments', [
+      ['target_id' => 'http://publications.europa.eu/resource/authority/corporate-body/ABEC'],
+      ['target_id' => 'http://publications.europa.eu/resource/authority/corporate-body/AASM'],
+    ])->save();
+    $this->drupalGet($node->toUrl());
+
+    $details_expected_values['items'][2] = [
+      'label' => 'Related departments',
+      'body' => 'Audit Board of the European Communities | Associated African States and Madagascar',
+    ];
+    $field_list_assert->assertPattern($details_expected_values, $content_items[0]->getHtml());
+
+    // Assert single Country field.
+    $node->set('oe_publication_country', 'http://publications.europa.eu/resource/authority/country/GBR');
+    $node->save();
+    $this->drupalGet($node->toUrl());
+
+    $details_expected_values['items'][] = [
+      'label' => 'Country',
+      'body' => 'United Kingdom',
+    ];
+    $field_list_assert->assertPattern($details_expected_values, $content_items[0]->getHtml());
+
+    // Assert multiple Country field.
+    $node->set('oe_publication_country', [
+      ['target_id' => 'http://publications.europa.eu/resource/authority/country/GBR'],
+      ['target_id' => 'http://publications.europa.eu/resource/authority/country/FRA'],
+    ])->save();
+    $this->drupalGet($node->toUrl());
+
+    $details_expected_values['items'][3] = [
+      'label' => 'Countries',
+      'body' => 'United Kingdom, France',
+    ];
+    $field_list_assert->assertPattern($details_expected_values, $content_items[0]->getHtml());
+
+    // Assert Body text field.
+    $node->set('body', 'Publication body text')->save();
+    $this->drupalGet($node->toUrl());
+
+    $inpage_nav_expected_values['list'] = [
+      ['label' => 'Details', 'href' => '#details'],
+      ['label' => 'Description', 'href' => '#description'],
+      ['label' => 'Files', 'href' => '#files'],
+    ];
+    $inpage_nav_assert->assertPattern($inpage_nav_expected_values, $navigation->getOuterHtml());
+
+    $content_items = $content->findAll('xpath', '/div');
+    $this->assertCount(3, $content_items);
+    $this->assertContentHeader($content_items[0], 'Details', 'details');
+    $this->assertContentHeader($content_items[1], 'Description', 'description');
+    $this->assertContentHeader($content_items[2], 'Files', 'files');
+
+    $body = $content_items[1]->findAll('css', '.ecl-row .ecl-col-12.ecl-col-md-8 .ecl-editor');
+    $this->assertCount(1, $body);
+    $this->assertEquals('Publication body text', $body[0]->getText());
```

Comment:
```suggestion
    $this->assertCount(1, $body);
    $body = reset($body);
    $this->assertEquals('Publication body text', $body->getText());
```

Reply:
The array has only one element so if you use reset it gets that one element and in the assert you don’t have to use an array but an element directly. I guess it's the same if you use it like this. Fine by me.

Reply:
But it is on one line shorter and might be a little bit faster.